### PR TITLE
[Merged by Bors] - feat(Order/Cover): add lemmas about `WithTop`/`WithBot`

### PR DESCRIPTION
**Pull request by urkud**
---
- [x] depends on: #15479

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Cover.lean [hunk 1]
```diff
@@ -522,3 +523,45 @@ theorem covBy_iff : x ⋖ y ↔ x.1 ⋖ y.1 ∧ x.2 = y.2 ∨ x.2 ⋖ y.2 ∧ x.
...
+@[simp] lemma coe_covBy_coe : (a : WithTop α) ⋖ b ↔ a ⋖ b :=
+  Set.OrdConnected.apply_covBy_apply_iff OrderEmbedding.withTopCoe <| by
+    simp [WithTop.range_coe, ordConnected_Iio]
+
+@[simp] lemma coe_covBy_top : (a : WithTop α) ⋖ ⊤ ↔ IsMax a := by
+  simp only [covBy_iff_Ioo_eq, ← image_coe_Ioi, coe_lt_top, image_eq_empty,
+    true_and, Ioi_eq_empty_iff]
+
+@[simp] lemma coe_wcovBy_top : (a : WithTop α) ⩿ ⊤ ↔ IsMax a := by
+  simp only [wcovBy_iff_Ioo_eq, ← image_coe_Ioi, le_top, image_eq_empty, true_and, Ioi_eq_empty_iff]
```
Comment (YaelDillies) [inline Mathlib/Order/Cover.lean:544]: Can these be `norm_cast`? Same below

File: Mathlib/Order/Interval/Set/Basic.lean [hunk 2]
```diff
@@ -566,11 +566,14 @@ theorem _root_.IsTop.Iic_eq (h : IsTop a) : Iic a = univ :=
...
+theorem Ioi_eq_empty_iff : Ioi a = ∅ ↔ IsMax a := by
+  simp only [isMax_iff_forall_not_lt, eq_empty_iff_forall_not_mem, mem_Ioi]
 
-theorem _root_.IsMin.Iio_eq (h : IsMin a) : Iio a = ∅ :=
-  eq_empty_of_subset_empty fun _ => h.not_lt
+alias ⟨_, _root_.IsMax.Ioi_eq⟩ := Ioi_eq_empty_iff
+
+theorem Iio_eq_empty_iff : Iio a = ∅ ↔ IsMin a := Ioi_eq_empty_iff (α := αᵒᵈ)
+
+alias ⟨_, _root_.IsMin.Iio_eq⟩ := Iio_eq_empty_iff
```
Comment (YaelDillies) [inline Mathlib/Order/Interval/Set/Basic.lean:576]:
```suggestion
theorem Iio_eq_empty_iff : Iio a = ∅ ↔ IsMin a := Ioi_eq_empty_iff (α := αᵒᵈ)

alias ⟨_, _root_.IsMax.Ioi_eq⟩ := Ioi_eq_empty_iff
alias ⟨_, _root_.IsMin.Iio_eq⟩ := Iio_eq_empty_iff
```

File: Mathlib/Order/Cover.lean [hunk 3]
```diff
@@ -522,3 +523,45 @@ theorem covBy_iff : x ⋖ y ↔ x.1 ⋖ y.1 ∧ x.2 = y.2 ∨ x.2 ⋖ y.2 ∧ x.
   exact mk_covBy_mk_iff
 
 end Prod
+
+namespace WithTop
+
+variable [Preorder α] {a b : α}
+
+@[simp] lemma coe_wcovBy_coe : (a : WithTop α) ⩿ b ↔ a ⩿ b :=
```
Comment (urkud) [inline Mathlib/Order/Cover.lean:531]:
```suggestion
@[simp, norm_cast] lemma coe_wcovBy_coe : (a : WithTop α) ⩿ b ↔ a ⩿ b :=
```

File: Mathlib/Order/Cover.lean [hunk 4]
```diff
@@ -522,3 +523,45 @@ theorem covBy_iff : x ⋖ y ↔ x.1 ⋖ y.1 ∧ x.2 = y.2 ∨ x.2 ⋖ y.2 ∧ x.
...
+  Set.OrdConnected.apply_wcovBy_apply_iff OrderEmbedding.withTopCoe <| by
+    simp [WithTop.range_coe, ordConnected_Iio]
+
+@[simp] lemma coe_covBy_coe : (a : WithTop α) ⋖ b ↔ a ⋖ b :=
```
Comment (urkud) [inline Mathlib/Order/Cover.lean:535]:
```suggestion
@[simp, norm_cast] lemma coe_covBy_coe : (a : WithTop α) ⋖ b ↔ a ⋖ b :=
```

File: Mathlib/Order/Cover.lean [hunk 5]
```diff
@@ -522,3 +523,45 @@ theorem covBy_iff : x ⋖ y ↔ x.1 ⋖ y.1 ∧ x.2 = y.2 ∨ x.2 ⋖ y.2 ∧ x.
...
+@[simp] lemma coe_wcovBy_top : (a : WithTop α) ⩿ ⊤ ↔ IsMax a := by
+  simp only [wcovBy_iff_Ioo_eq, ← image_coe_Ioi, le_top, image_eq_empty, true_and, Ioi_eq_empty_iff]
+
+end WithTop
+
+namespace WithBot
+
+variable [Preorder α] {a b : α}
+
+@[simp] lemma coe_wcovBy_coe : (a : WithBot α) ⩿ b ↔ a ⩿ b :=
```
Comment (urkud) [inline Mathlib/Order/Cover.lean:552]:
```suggestion
@[simp, norm_cast] lemma coe_wcovBy_coe : (a : WithBot α) ⩿ b ↔ a ⩿ b :=
```

File: Mathlib/Order/Cover.lean [hunk 6]
```diff
@@ -522,3 +523,45 @@ theorem covBy_iff : x ⋖ y ↔ x.1 ⋖ y.1 ∧ x.2 = y.2 ∨ x.2 ⋖ y.2 ∧ x.
...
+  Set.OrdConnected.apply_wcovBy_apply_iff OrderEmbedding.withBotCoe <| by
+    simp [WithBot.range_coe, ordConnected_Ioi]
+
+@[simp] lemma coe_covBy_coe : (a : WithBot α) ⋖ b ↔ a ⋖ b :=
```
Comment (urkud) [inline Mathlib/Order/Cover.lean:556]:
```suggestion
@[simp, norm_cast] lemma coe_covBy_coe : (a : WithBot α) ⋖ b ↔ a ⋖ b :=
```

File: Mathlib/Order/Cover.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Order/Cover.lean]: I added some `norm_cast` tags.

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**